src/spatial/mod.rs

```diff
@@ -14,21 +14,21 @@ pub(crate) const FLOAT_ERROR_TOLERANCE: f32 = 0.00001;
     derive(serde::Serialize, serde::Deserialize)
 )]
 pub(crate) struct Cube {
-    pub(crate) min_position: V3c<u32>,
-    pub(crate) size: u32,
+    pub(crate) min_position: V3c<f32>,
+    pub(crate) size: f32,
```

Comment:
Huh, what motivated this change?

Reply:
Basically this flamegraph analysis: 
![image](https://github.com/davids91/shocovox/assets/4211067/07720845-56c3-4943-92cb-1456bc405e5e)
A mentionable amount of compute went into converting this back and forth.. In the end this change made the CPU raycast benchmarks 8-15% faster

Reply:
Fascinating, I wonder if the same would happen with just a u32 or a f32... I mean, both are just 32 bits wide, and what those bits encode is up to interpretation...

Reply:
I'm not sure if the size match would guarantee that adequate conversion is lightweight though. 

On the same note: https://en.wikipedia.org/wiki/Fast_inverse_square_root 
Uses similar ideas!